Source/Source/Utilities/FoodUtility.cs

```diff
@@ -0,0 +1,48 @@
+﻿using RimWorld;
+using Verse;
+
+namespace Hospitality.Utilities
+{
+    public static class FoodUtility
+    {
+        public static bool GuestCanSatisfyFoodNeed(Pawn guest)
+        {
+            //Check Inventory
+            var inventoryFood = RimWorld.FoodUtility.BestFoodInInventory(guest, minFoodPref: FoodPreferability.RawTasty);
+            if (inventoryFood != null) return true;
+
+            //Search FoodSource
+            if (RimWorld.FoodUtility.TryFindBestFoodSourceFor(guest, guest, false, out var foodSource, out var foodDef, true, false, false, false, false, false, false, false, false, FoodPreferability.RawTasty))
+            {
+                if (foodSource != null && foodDef != null) return true;
+            }
+
+            return false;
+        }
+
+        public static bool TryPayForFood(Pawn buyerGuest, Building_NutrientPasteDispenser dispenser)
+        {
+            var vendingMachine = dispenser.TryGetComp<CompVendingMachine>();
+            if (vendingMachine.IsActive() && dispenser.CanDispenseNow)
+            {
+                if (vendingMachine.IsFree) return true;
+
+                if (!vendingMachine.CanAffordFast(buyerGuest, out Thing silver)) return false;
+
+                vendingMachine.ReceivePayment(buyerGuest.inventory.innerContainer, silver);
+                return true;
+            }
+            return false;
+        }
+
+        public static bool WillConsume(Pawn pawn, ThingDef foodDef)
```

Comment:
Has this function been moved or is it double? It looks really familiar...

Reply:
It's based of the Gastronomy.RestaurantStock.WillConsume() method, I found it while looking into how to make DiningSpots a viable option for available food sources. I adjusted it a bit as vending machines probably never have to dispense drugs.
Added a general check to consider restrictions too (even if they arent configurable for guests).